tests/test_sequence.py

```diff
@@ -1449,22 +1452,39 @@ def test_eom_mode(reg, mod_device, custom_buffer_time, patch_plt_show):
     ]
 
     pulse_duration = 100
-    seq.add_eom_pulse("ch0", pulse_duration, phase=0.0)
+    seq.add_eom_pulse(
+        "ch0",
+        pulse_duration,
+        phase=0.0,
+        correct_phase_drift=correct_phase_drift,
+    )
     first_pulse_slot = seq._schedule["ch0"].last_pulse_slot()
     assert first_pulse_slot.ti == delay_slot.tf
     assert first_pulse_slot.tf == first_pulse_slot.ti + pulse_duration
-    eom_pulse = Pulse.ConstantPulse(pulse_duration, amp_on, detuning_on, 0.0)
+    phase = detuning_off * first_pulse_slot.ti * 1e-3 * correct_phase_drift
+    eom_pulse = Pulse.ConstantPulse(pulse_duration, amp_on, detuning_on, phase)
     assert first_pulse_slot.type == eom_pulse
     assert not seq._schedule["ch0"].is_detuned_delay(eom_pulse)
 
     # Check phase jump buffer
-    seq.add_eom_pulse("ch0", pulse_duration, phase=np.pi)
+    phase_ = np.pi
+    seq.add_eom_pulse(
+        "ch0",
+        pulse_duration,
+        phase=phase_,
+        correct_phase_drift=correct_phase_drift,
+    )
     second_pulse_slot = seq._schedule["ch0"].last_pulse_slot()
     phase_buffer = (
         eom_pulse.fall_time(ch0_obj, in_eom_mode=True)
         + seq.declared_channels["ch0"].phase_jump_time
     )
     assert second_pulse_slot.ti == first_pulse_slot.tf + phase_buffer
+    # Corrects the phase acquired during the phase buffer
+    phase_ += detuning_off * phase_buffer * 1e-3 * correct_phase_drift
```

Comment:
Coherent with the two minuses seen above... But shouldn't `detuning_off*phase_buffer*1e3` be substracted from the current phase ?